includes/Core/Authentication/Authentication.php

```diff
@@ -405,6 +405,25 @@ function ( $user_id ) {
 			add_action( 'googlesitekit_authorize_user', $set_initial_version );
 			add_action( 'googlesitekit_reauthorize_user', $set_initial_version );
 		}
+
+		$maybe_refresh_token = function() {
+			$token = $this->token->get();
```

Comment:
Maybe a good idea to add a permission check here for whether the user can generally authenticate? Not super-critical, but I think that's a good pattern. A person without permissions shouldn't have a token set anyway, but it could be relevant e.g. if your permission is downgraded from admin to editor.

Also, maybe worth checking for whether Site Kit credentials are set as well, otherwise this won't work.

Reply:
Added permissions and credentials check.